src/coreclr/vm/gcheaputilities.h

```diff
@@ -26,6 +26,14 @@ GVAL_DECL(gc_alloc_context, g_global_alloc_context);
 }
 #endif // !DACCESS_COMPILE
 
+// on MP systems, each thread has its own allocation chunk so we can avoid
+// lock prefixes and expensive MP cache snooping stuff
+#ifdef _MSC_VER
+EXTERN_C __declspec(selectany) __declspec(thread) gc_alloc_context t_thread_alloc_context;
+#else
+EXTERN_C __thread gc_alloc_context t_thread_alloc_context;
```

Comment:
I was about to see what it would take to move m_fPreemptiveGCDisabled and m_pFrame to be on this same plan (inline thread static - to save an indirection in PInvoke transitions). It would help to have all these key inline thread statics together, similar to how it is done in naot. Can we turn this into a structure like `ThreadBuffer` and make `gc_alloc_context ` to be the first field in it? We can then move more into that structure.

Reply:
I can do that. Would you like me to do it in this PR or in a follow-up?

I'd prefer to do it in a follow-up when I mess with the box helper as there won't be any assembly changes I need to make then (as the thread-specific alloc-context won't be referenced in any hand-written CoreCLR assembly helpers at that point)

Reply:
Either way is fine with me.